### ISSUE TITLE
[a11y] General a11y criteria

### ISSUE THREAD

**Issue by ThomasEverding** [open-pioneer/trails-openlayers-base-packages#48]
a11y criteria for all functions and components:

 - all images must have alt-texts.
 - sections must be separated by headings, these headings must be marked up
 - the text-based sections must be screen-readable
 - the html syntax contains no errors
 - the color contrast must be at least 4.5:1 for normal text (AA level)
   - the color contrast should be at least 7:1 for normal text (AAA level)
 - the color contrast must be at least 3:1 for larger text (>= 24px) (AA level)
   - the color contrast should be at least 4.5:1 for larger text (>= 24px) (AAA level)
 - all form controls must have labels, these labels must be marked up
 - all form controls must be accessible by keyboard
 - the keyboard focus must be visible
 - the keyboard focus order must be logical within a form / component
 - all elements of a form / component in the html tree must be ordered in a logical and understandable way
 - it must be clearly indicated if form fields are required/mandatory; this may not be achieved by color alone
 - necessary instructions must be shown before they are needed
 - form errors must be clearly highlighted and guidance to fix the error shall be given; the errors shall be easily findable
 - the form / component GUI must be usable with a 200% zoom without content overlaps
 - The default human language of each Web page can be programmatically determined

Sources:
 - con terra a11y quick check
 - BITV Test results (https://studio.bitvtest.de/)

out of scope
 - simple language / NLS
